src/nuclio/functions/function-collapsing-row/function-collapsing-row.component.js

```diff
@@ -265,24 +265,38 @@
 
         /**
          * Deletes function from functions list
+         * @param {Object} functionItem
          * @returns {Promise}
          */
-        function deleteFunction() {
+        function deleteFunction(functionItem) {
             if (lodash.isEmpty(apiGateways)) {
                 ctrl.isSplashShowed.value = true;
 
-                return ctrl.handleDeleteFunction({functionData: ctrl.function.metadata})
+                return ctrl.handleDeleteFunction({
+                    functionData: functionItem ? functionItem.metadata : ctrl.function.metadata
```

Comment:
Just to be consistent with the rest of the codebase, we always use boolean expression for conditionals.
Here are a few suggestions:
```suggestion
                    functionData: lodash.isNil(functionItem) ? ctrl.function.metadata : functionItem.metadata
```
Or, not having the conditional at all:

```suggestion
                    functionData: lodash.defaultTo(functionItem, ctrl.function).metadata
```
This is my least favorite one:
```suggestion
                    functionData: lodash.get(functionItem, 'metadata', ctrl.function.metadata)
```